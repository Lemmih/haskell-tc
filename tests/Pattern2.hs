module Pattern2 where

data Maybe a = Nothing | Just a

fromMaybe def Nothing  = def
fromMaybe def (Just a) = a
