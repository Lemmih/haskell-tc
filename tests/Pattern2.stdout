Bindings:
:: * â†’  *
  data [4mMaybe[0m a = Nothing | Just a
:: Î›a. Pattern2.Maybe a
  data Maybe a = [4mNothing[0m | Just a
:: Î›a. a â†’  Pattern2.Maybe a
  data Maybe a = Nothing | [4mJust[0m a
:: âˆ€ v7. v7 â†’  Pattern2.Maybe v7 â†’  v7
  [4mfromMaybe def Nothing  = def[0m
  [4mfromMaybe def (Just a) = a[0m
:: v7
  fromMaybe [4mdef[0m Nothing  = def
:: v7
  fromMaybe [4mdef[0m (Just a) = a
:: v7
  fromMaybe def (Just [4ma[0m) = a

Proofs:
coercion: (âˆ€ a. Pattern2.Maybe a) @ v7
  fromMaybe def [4mNothing[0m  = def
coercion: (âˆ€ a. a â†’  Pattern2.Maybe a) @ v7
  fromMaybe def ([4mJust[0m a) = a
coercion: v7
  fromMaybe def Nothing  = [4mdef[0m
coercion: v7
  fromMaybe def (Just a) = [4ma[0m
